src/components/Reactions/Reactions.tsx

```diff
@@ -14,16 +15,48 @@ type FilledComment = Comment & {
   };
 };
 
+type FilledLike = Like & {
+  author: {
+    email: string;
+  };
+};
+
 type Props = {
   comments: FilledComment[];
+  likes: FilledLike[];
+  onSubmitNewLike: Function;
 };
 
-const Reactions: React.FC<Props> = ({comments}: Props) => {
+const Reactions: React.FC<Props> = (props: Props) => {
+  const { loggedUser } = useContext(AuthContext);
+  const [ alreadyLiked, setAlreadyLiked ] = useState(undefined);
+
+  const likes = props.likes ? props.likes.length : 0;
+
+  useEffect(() => {
+    if(loggedUser) {
+      setAlreadyLiked(
+        props.likes.find(like => like.author.email === loggedUser?.email)
+      );
+    };
+  }, [props.likes, loggedUser])
+
   return(
     <div className={styles.reactions}>
+      <div 
+        className={styles.actions}
+        onClick={e => {
+          e.preventDefault();
+          props.onSubmitNewLike(alreadyLiked ? alreadyLiked.id : null);
+        }}
+      >
+        <BiLike className={alreadyLiked ? styles.unliked : styles.liked} />
```

Comment:
This loggic doesnt seem right. if i already liked my style will be unliked? what are you trying to say here? try changing it to something like: alreadyLiked ? styles.unlike : styles.like or  styles.alreadyLiked : styles.neverLiked